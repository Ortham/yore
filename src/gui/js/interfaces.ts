export interface Coordinates {
  latitude: number;
  longitude: number;
}

export interface Photo {
  path: string;
  location?: Location; // eslint-disable-line no-restricted-globals
  error?: string;
  height?: number;
  width?: number;
  loaded?: boolean;
}

interface Location {
  Existing?: Coordinates;
  Suggested?: [Coordinates, LocationAccuracy];
}

export interface LocationAccuracy {
  meters: number;
  seconds: number;
}

export interface LocationResponse {
  location?: Location; // eslint-disable-line no-restricted-globals
  error?: string;
}
