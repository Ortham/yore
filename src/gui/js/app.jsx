import 'babel-polyfill';
import 'whatwg-fetch';
import React from 'react';
import ReactDOM from 'react-dom';
import Page from './page';
import { getPhotos, getRootPath } from './requests';

function getInitialState() {
  const state = {
    rootPath: undefined,
    filterPhotos: false,
    currentPhoto: undefined,
    photos: []
  };

  return getRootPath()
    .then(responseBody => {
      state.rootPath = responseBody.rootPath;

      return getPhotos();
    })
    .then(photos => {
      state.photos = photos;

      return state;
    });
}

getInitialState().then(state => {
  ReactDOM.render(
    <Page rootPath={state.rootPath} photos={state.photos} />,
    document.getElementById('root')
  );
});
