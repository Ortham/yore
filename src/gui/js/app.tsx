import '@babel/polyfill';
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import 'whatwg-fetch';
import { Page, PageProps } from './page';
import {
  getInterpolate,
  getLocationHistoryPath,
  getPhotos,
  getRootPath,
  getLocations
} from './requests';
import { updatePhotoLocations } from './photo';

function getInitialState() {
  const state: PageProps = {
    interpolate: false,
    locationHistoryPath: undefined,
    photos: [],
    rootPath: undefined
  };

  return getRootPath()
    .then(responseBody => {
      state.rootPath = responseBody.rootPath || '';

      return getPhotos();
    })
    .then(photos =>
      getLocations(0, photos.length).then(locations =>
        updatePhotoLocations(photos, locations)
      )
    )
    .then(photos => {
      state.photos = photos;

      return getInterpolate();
    })
    .then(responseBody => {
      state.interpolate = responseBody.interpolate;

      return getLocationHistoryPath();
    })
    .then(responseBody => {
      state.locationHistoryPath = responseBody.locationHistoryPath || '';

      return state;
    });
}

getInitialState().then(state => {
  ReactDOM.render(<Page {...state} />, document.getElementById('root'));
});
