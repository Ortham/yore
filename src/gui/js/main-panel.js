import React from 'react';
import MapArea from './map-area';
import {locationDescription, hasSuggestion} from './photo';

export default function MainPanel(props) {
    if (props.photo) {
        return (
            <main>
                <section>
                    <img src={props.photo.src} />
                    <MapArea photo={props.photo} />
                </section>
                <footer>
                    {locationDescription(props.photo)}
                    <br />
                    <button
                        disabled={!hasSuggestion(props.photo)}
                        onClick={props.handleSuggestionApply} >
                        Apply
                    </button>
                    <button
                        disabled={!hasSuggestion(props.photo)}
                        onClick={props.handleSuggestionDiscard} >
                        Discard
                    </button>
                </footer>
            </main>
        );
    } else {
        return (
            <main>
                <section>No photo selected.</section>
                <footer>
                    <br />
                    <button disabled>Apply</button>
                    <button disabled>Discard</button>
                </footer>
            </main>
        );
    }
}
