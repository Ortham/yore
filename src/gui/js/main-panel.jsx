import React from 'react';
import PropTypes from 'prop-types';
import MapArea from './map-area';
import { locationDescription, hasSuggestion } from './photo';

export default function MainPanel(props) {
  if (props.photo) {
    return (
      <div id="main">
        <section>
          <img src={props.photo.src} alt="Selected" />
          <MapArea photo={props.photo} />
        </section>
        <footer>
          <div>{locationDescription(props.photo)}</div>
          <div>
            <button
              disabled={!hasSuggestion(props.photo)}
              onClick={props.handleSuggestionApply}
            >
              Apply
            </button>
            <button
              disabled={!hasSuggestion(props.photo)}
              onClick={props.handleSuggestionDiscard}
            >
              Discard
            </button>
          </div>
        </footer>
      </div>
    );
  }
  return (
    <div id="main">
      <section>No photo selected.</section>
      <footer>
        <div />
        <div>
          <button disabled>Apply</button>
          <button disabled>Discard</button>
        </div>
      </footer>
    </div>
  );
}

MainPanel.propTypes = {
  photo: PropTypes.shape({
    src: PropTypes.string
  }),
  handleSuggestionApply: PropTypes.func.isRequired,
  handleSuggestionDiscard: PropTypes.func.isRequired
};

MainPanel.defaultProps = {
  photo: undefined
};
