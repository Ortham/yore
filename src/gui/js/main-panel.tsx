import * as React from 'react';
import { Photo } from './interfaces';
import { MapArea } from './map-area';
import { hasSuggestion, locationDescription } from './photo';

export interface MainPanelProps {
  photo?: Photo;
  handleSuggestionApply: () => Promise<void>;
  handleSuggestionDiscard: () => void;
}

export function MainPanel(props: MainPanelProps) {
  if (props.photo) {
    return (
      <div id="main">
        <section>
          <img src={props.photo.src} alt="Selected" />
          <MapArea photo={props.photo} />
        </section>
        <footer>
          <div>{locationDescription(props.photo)}</div>
          <div>
            <button
              disabled={!hasSuggestion(props.photo)}
              onClick={props.handleSuggestionApply}
            >
              Apply
            </button>
            <button
              disabled={!hasSuggestion(props.photo)}
              onClick={props.handleSuggestionDiscard}
            >
              Discard
            </button>
          </div>
        </footer>
      </div>
    );
  }
  return (
    <div id="main">
      <section>No photo selected.</section>
      <footer>
        <div />
        <div>
          <button disabled>Apply</button>
          <button disabled>Discard</button>
        </div>
      </footer>
    </div>
  );
}
