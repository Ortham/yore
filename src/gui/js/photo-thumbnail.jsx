import React from 'react';
import PropTypes from 'prop-types';
import { chooseIcon } from './photo';

export default function PhotoThumbnail(props) {
  return (
    <div
      style={props.style}
      className="photo"
      onClick={props.handleSelect}
      onKeyUp={props.handleSelect}
      role="option"
      aria-selected={props.isSelected}
      tabIndex="0"
    >
      <img src={props.photo.src} title={props.photo.path} alt="thumbnail" />
      {chooseIcon(props.photo)}
    </div>
  );
}

PhotoThumbnail.propTypes = {
  style: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
  isSelected: PropTypes.bool.isRequired,
  handleSelect: PropTypes.func.isRequired,
  photo: PropTypes.shape({
    path: PropTypes.string,
    src: PropTypes.string
  }).isRequired
};
