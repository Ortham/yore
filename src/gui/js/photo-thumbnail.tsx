import * as React from 'react';
import { Photo } from './interfaces';
import { chooseIcon } from './photo';

export interface PhotoThumbnailProps {
  isSelected: boolean;
  photo: Photo;
  src: string;
  loadedClassName: string;
  handleSelect: () => void;
}

export function PhotoThumbnail(props: PhotoThumbnailProps) {
  return (
    <div
      className="photo"
      onClick={props.handleSelect}
      onKeyUp={props.handleSelect}
      role="option"
      aria-selected={props.isSelected}
      tabIndex={0}
    >
      <img
        src={props.src}
        title={props.photo.path}
        alt="thumbnail"
        onLoad={evt => {
          if (evt.target instanceof HTMLElement) {
            evt.target.parentElement.classList.add(props.loadedClassName);
          }
        }}
      />
      {chooseIcon(props.photo)}
    </div>
  );
}
