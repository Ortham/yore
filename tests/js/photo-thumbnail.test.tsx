import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { PhotoThumbnail } from '../../src/gui/js/photo-thumbnail';

describe('PhotoThumbnail', () => {
  let mockHandleSelect: jest.Mock;
  let thumbnail: renderer.ReactTestRendererJSON;

  beforeAll(() => {
    mockHandleSelect = jest.fn();

    thumbnail = renderer
      .create(
        <PhotoThumbnail
          key="path"
          isSelected={false}
          handleSelect={mockHandleSelect}
          photo={{
            path: 'path'
          }}
          loadedClassName="pig-loaded"
          src="/thumbnail?path=path&maxWidth=300&maxHeight=300"
        />
      )
      .toJSON();
  });

  beforeEach(() => {
    mockHandleSelect.mockClear();
  });

  test('renders a div with an image and an icon', () => {
    expect(thumbnail).toMatchSnapshot();
  });

  test('clicking on a PhotoThumbnail should call its handleSelect callback', () => {
    thumbnail.props.onClick();
    expect(mockHandleSelect.mock.calls.length).toBe(1);
  });

  test('a key-up on a PhotoThumbnail should call its handleSelect callback', () => {
    thumbnail.props.onKeyUp();
    expect(mockHandleSelect.mock.calls.length).toBe(1);
  });
});
