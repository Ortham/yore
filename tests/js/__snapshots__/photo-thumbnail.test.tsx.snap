// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoThumbnail renders a div with an image and an icon 1`] = `
<div
  aria-selected={false}
  className="photo"
  onClick={[MockFunction]}
  onKeyUp={[MockFunction]}
  role="option"
  tabIndex={0}
>
  <img
    alt="thumbnail"
    onLoad={[Function]}
    src="/thumbnail?path=path&maxWidth=300&maxHeight=300"
    title="path"
  />
</div>
`;
